plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx

```diff
@@ -77,14 +77,17 @@ export const EntitySwitch = (props: EntitySwitchProps) => {
         })
         .getElements()
         .flatMap<SwitchCaseResult>((element: ReactElement) => {
-          // If the entity is missing or there is an error, render nothing
-          if (!entity) {
```

Comment:
as per discussions elsewhere - i still think that maybe we need to do `if (loading && !entity) { return [] }` to make sure that during _initial_ load, the fallback isn't shown.